src/dividat-driver/firmware/main.go

```diff
@@ -100,35 +108,40 @@ func Update(parentCtx context.Context, image io.Reader, deviceSerial *string, co
 			// We do need to rediscover, as the legacy device may still just have
 			// restarted into the bootloader and obtained a new IP address.
 			ctx, cancel := context.WithTimeout(parentCtx, 60*time.Second)
-			legacyDiscoveredAddr, err := discover("_sensoControl._tcp", deviceSerial, ctx)
+			legacyDiscoveredAddr, err := discover("_sensoControl._tcp", deviceSerial, ctx, onProgress)
```

Comment:
We could check the `mode` txt entry here to see if we discovered an actual bootloader.